### PR TITLE
[IMP] website_forum: remove karma restriction, now users can post as a link or image

### PR DESCRIPTION
**Pull request by rim-odoo**
- Task: https://www.odoo.com/web#id=8712&view_type=form&model=project.task&menu_id=3942&action=327
- Pad: https://pad.odoo.com/p/r.Mo2LPzffbGk1mjxD

### CODE REVIEW

File: addons/website/static/lib/summernote/src/js/EventHandler.js [hunk 1]
```diff
@@ -420,7 +420,7 @@ define([
           $btn.parents('.popover').hide();
         }
 
-        if ($.isFunction($.summernote.pluginEvents[eventName])) {
+        if ($.isFunction($.summernote.pluginEvents[eventName]) && eventName != "formatBlock") {
```
Comment (tde-banana-odoo) [inline addons/website/static/lib/summernote/src/js/EventHandler.js]: No modification in summernote; we have to stick on the lib.

File: addons/website_forum/models/forum.py [hunk 2]
```diff
@@ -110,6 +110,7 @@ def _get_default_faq(self):
     karma_retag = fields.Integer(string='Change question tags', default=75)
     karma_flag = fields.Integer(string='Flag a post as offensive', default=500)
     karma_dofollow = fields.Integer(string='Disabled links', help='If the author has not enough karma, a nofollow attribute is added to links', default=500)
+    karma_post_image = fields.Integer(string='Karma to post either an image or link (question, answer, edit).', default=30)
```
Comment (tde-banana-odoo) [inline addons/website_forum/models/forum.py]: karma_editor_image, + move with other editor-related posts.

File: addons/website_forum/models/forum.py [hunk 3]
```diff
@@ -137,6 +138,25 @@ def _tag_to_write_vals(self, tags=''):
         post_tags.insert(0, [6, 0, existing_keep])
         return post_tags
 
+    @api.model
+    def _keep_tag_vals(self, tags=''):
```
Comment (tde-banana-odoo) [inline addons/website_forum/models/forum.py]: Hugh ?

File: addons/website_forum/controllers/main.py [hunk 4]
```diff
@@ -276,6 +277,37 @@ def forum_post(self, forum, post_type=None, **post):
     def post_create(self, forum, post_parent=None, post_type=None, **post):
         if not request.session.uid:
             return login_redirect()
+
+        if post.get('content', None):
+            validate_karma_content = forum._ckeditor_content_filter(forum, post.get('content', None))
```
Comment (tde-banana-odoo) [inline addons/website_forum/controllers/main.py]: The check should be done directly in the create / write, in a method that check the content that is posted (called in both methods). This method should raise a KarmaError. Indeed the check is something we intend to have at model-level; the controller has nothing to do with content control.

You should define a json controller that check the content before posting, so no need to define a new template or to redraw the whole page.

File: addons/website/static/lib/summernote/src/js/EventHandler.js (hunk 1 above)
Comment (jpa-odoo) [inline addons/website/static/lib/summernote/src/js/EventHandler.js]: I compare <i>SummerNote master</i> and <i>Odoo SummerNote</li> lot's of override or changes are apply on library files,

View following screenshot,
summernote_eventhandler_js_cropped

This <i>if....else..</i> condition block portion copy from summernote master to our library <i>formatblock</i> functionality worked, but <i>orderlist</i> and <i>unorderlist</i> functionality are break. Can you suggest what can I do for this case. thank you,

File: addons/website_forum/models/forum.py (hunk 2 above)
Comment (jpa-odoo) [inline addons/website_forum/models/forum.py]: ok,

File: addons/website/static/lib/summernote/src/js/EventHandler.js (hunk 1 above)
Comment (Gorash) [inline addons/website/static/lib/summernote/src/js/EventHandler.js]: Please, no modification in summernote (except my changes) as I undertake the remains of the library and create the merge proposal to have our feature and fix in the lib.
If you want to change formatBlock you must look in our files who use $.summernote.pluginEvents. But for this task I think no need to change comportment.

Comment (jpa-odoo) [inline addons/website/static/lib/summernote/src/js/EventHandler.js]: Ok, I revert this changes and make another task for fix code, quote formatting plugin.
